src/containers/MyCart/MyCart.tsx

```diff
@@ -1,35 +1,28 @@
 import React from "react";
-import { Image, ScrollView, TouchableOpacity, View, ViewStyle } from "react-native";
-import { images } from "../../assets/images";
-import { Button, FontWeights, Header, Typography } from "../../components";
+import { SafeAreaView, ViewStyle } from "react-native";
+import { ScrollView } from "react-native-gesture-handler";
+import { Header } from "../../components";
+import { useAuthContext } from "../../context";
 import { NavigationProp } from "../../types";
+import { MyCartLogin } from "./components";
+import { MyCartEmpty } from "./components/MyCartEmpty";
 import { styles } from "./MyCart.styles";
 
 type MyCartProps = {
   style?: ViewStyle | ViewStyle[];
 } & NavigationProp;
 
 export const MyCart = ({ style, navigation }: MyCartProps) => {
+  const { loginData } = useAuthContext();
   return (
     <>
-      <Header type="main" navigation={navigation} />
-      <View style={[styles.myCartContainer, style]}>
-        <Image source={images.avatar} />
-        <Typography
-          variant="h6"
-          style={{ fontWeight: FontWeights.fontWeightBold, marginTop: 16 }}
-          color="gray"
-        >
-          Login First!
-        </Typography>
-        <Typography color="gray" style={{ marginTop: 4 }}>
-          Login first to view your cart
-        </Typography>
-        <Button title="LOGIN NOW" style={{ marginTop: 32 }} fullWidth />
-        <TouchableOpacity style={{ marginTop: 24 }}>
-          <Typography color="blue">New here? Sign Up</Typography>
-        </TouchableOpacity>
-      </View>
+      <SafeAreaView>
+        <Header />
+      </SafeAreaView>
+      <ScrollView contentContainerStyle={[styles.myCartContainer, style]}>
+        {!loginData && <MyCartLogin navigation={navigation} />}
```

Comment:
better to use conditional render on the navigation level
` loginData ? <Screen name='MyCart'/> : <Screen name='pleaseLogin'/>`

Reply:
Its not better in this case cause there would be a 3rd case with filled cart and it would cause nested ternary
Its quite readable as it is